agent/config/config.go

```diff
@@ -269,6 +272,11 @@ type Telemetry struct {
 	CirconusBrokerSelectTag string `hcl:"circonus_broker_select_tag,optional"`
 }
 
+type HighAvailability struct {
+	Enable   bool   `hcl:"high_availability"`
+	LockPath string `hcl:"lock_path,optional"  json:"-"`
+}
```

Comment:
Do we want to expose more options, such as the TTL and LockDelay? 

Checking with the Nomad config docs, we seem to prefer `enabled` which we might want to follow.

The HCL struct tag on the `enable` parameter seems wrong.